gson/src/main/java/com/google/gson/NewlineStyle.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (C) 2022 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.gson;
+
+/**
+ * An enumeration that defines the kind of newline to use for serialization.
+ *
+ * @see <a href="https://en.wikipedia.org/wiki/Newline">Wikipedia Newline article</a>
+ *
+ * @since $next-version$
+ */
+public enum NewlineStyle {
+  /**
+   * Using this style will result in the same kind of newline that the current environment uses.
+   *
+   * <p>So it will produce {@code "\r\n"} when running on Windows, and {@code "\n"} when running
+   * on macOS &amp; Linux.</p>
```

Comment:
Maybe it would be good to recommend here to not use `CURRENT_OS` when exchanging the JSON data between systems.

Reply:
Done.

Also added a test (`NewlineTest.testNewlineInterop`) making sure that all
the combinations of parsers and inputs work, regardless the newline style.